addons/binding/org.openhab.binding.avmfritz/src/main/java/org/openhab/binding/avmfritz/BindingConstants.java

```diff
@@ -29,25 +29,44 @@
     public static final String SERIAL_NUMBER = "serialNumber";
     public static final String DEVICE_ID = "deviceId";
     public static final String BRIDGE_MODEL_NAME = "FRITZ!Box";
-    public static final String PL546E_MODEL_NAME = "FRITZ!Powerline";
+    public static final String POWERLINE_MODEL_NAME = "FRITZ!Powerline";
     public static final String THING_AIN = "ain";
 
     // List of main device types
     public static final String DEVICE_DECT210 = "FRITZ_DECT_210";
     public static final String DEVICE_DECT200 = "FRITZ_DECT_200";
     public static final String DEVICE_DECT100 = "FRITZ_DECT_Repeater_100";
     public static final String DEVICE_PL546E = "FRITZ_Powerline_546E";
+    public static final String DEVICE_PL1240E_STANDALONE = "FRITZ_Powerline_1240E_Solo";
+    public static final String DEVICE_PL1220E_STANDALONE = "FRITZ_Powerline_1220E_Solo";
+    public static final String DEVICE_PL1000E_STANDALONE = "FRITZ_Powerline_100E_Solo";
     public static final String DEVICE_PL546E_STANDALONE = "FRITZ_Powerline_546E_Solo";
-
+    public static final String DEVICE_PL540E_STANDALONE = "FRITZ_Powerline_540E_Solo";
+    public static final String DEVICE_PL530E_STANDALONE = "FRITZ_Powerline_530E_Solo";
+    public static final String DEVICE_PL510E_STANDALONE = "FRITZ_Powerline_510E_Solo";
+    
     // List of all Thing Type UIDs
     public final static ThingTypeUID BRIDGE_THING_TYPE = new ThingTypeUID(BINDING_ID, BRIDGE_FRITZBOX);
+    public final static ThingTypeUID POWERLINE_THING_TYPE = new ThingTypeUID(BINDING_ID, "Powerline_Solo");
 
     public final static ThingTypeUID DECT210_THING_TYPE = new ThingTypeUID(BINDING_ID, DEVICE_DECT210);
     public final static ThingTypeUID DECT200_THING_TYPE = new ThingTypeUID(BINDING_ID, DEVICE_DECT200);
     public final static ThingTypeUID DECT100_THING_TYPE = new ThingTypeUID(BINDING_ID, DEVICE_DECT100);
     public final static ThingTypeUID PL546E_THING_TYPE = new ThingTypeUID(BINDING_ID, DEVICE_PL546E);
+    public final static ThingTypeUID PL1240E_STANDALONE_THING_TYPE = new ThingTypeUID(BINDING_ID,
```

Comment:
Correct, but these constants are used to define the supported thing types later in this file.

If I understood the programming correctly the binding uses a xml-attribute "productname" of the interface to identify a supported device during discovery. This attribute contains the real description of the FRITZ! product (whitespaces are replaced by underscores).